R/edgeThresh.R

```diff
@@ -0,0 +1,461 @@
+#'     is the same as the component the receiving endpoint belongs to. Thus
+#'     the component maintenance is achieved in 2 steps (one query and one
+#'     comparison) for each receiving node, and thus 4 * Q * |E| steps are
+#'     required.
+#'
+#' @section Incrementing δ and Lmax:
+#'     After creating the permutated graph, the following procedure is run on
+#'     the graph for each Lmax starting from Lmin to the Lmax arguments.
+#'     Initially, δ values are chosen by finding min(|E|, 100) quantiles of
+#'     edge weight distribution, and used to delete edges from the permutation.
+#'     As soon as the largest component has size (ord) between Lmin and Lmax,
+#'     If the current permutation's ord values is less than Lmin (without
+#'     finding δ), the delta is skipped, and interpolated later on.
+#'
+#' @param EGG An EGG object - the input heat-equilibrated gene score graph.
+#' @param Lmax	Maximum order for the largest connected component in a random
+#' network. See details.
+#' @param Lmin Minimum order for the largest connected component. Defaults to 2.
+#' @param N	The number of permuted networks to produce for choosing a δ
+#' cutoff.
+#' @param Q	The number of edge-swapping permutations to produce one of the
+#' permuted networks is Q * |E|.
+#' @param silent Boolean. Default: FALSE. Whether or not to write
+#' progress information to console.
+#' @param writeLog Boolean. Default: TRUE Whether or not to log results.
+#'
+#' @return The Equilibrated Gene Graph EGG with δ values as additional
+#' graph metadata.
+#'
+#' @references
+#' @seealso
+#' @examples
+#'
+#' @export
+THRESH <- function(EGG,
+                   Lmax = max(getOptions("rete.minSubnetOrder")),
+                   Lmin = 2,
+                   N = 100,
+                   Q = 100,
+                   silent = FALSE,
+                   writeLog = TRUE) {
+    # ==== VALIDATIONS =========================================================
+
+    # Check if EGG is given
+    if (missing(EGG)) {
+        stop(paste0("Missing input EGG, with no defaults."))
+    }
+    # General parameter checks (TODO)
+    # cR <- character()
+    # cR <-
+    #     c(cR, .checkArgs(EGG,         like = getOptions("rete.EGGprototype")))
+    # cR <-
+    #     c(cR, .checkArgs(Lmax,        like = numeric(1), checkSize = TRUE))
+    # cR <-
+    #     c(cR, .checkArgs(N,           like = numeric(1), checkSize = TRUE))
+    # cR <-
+    #     c(cR, .checkArgs(Q,           like = numeric(1), checkSize = TRUE))
+    # cR <-
+    #     c(cR, .checkArgs(silent,      like = logical(1), checkSize = TRUE))
+    # cR <-
+    #     c(cR, .checkArgs(writeLog,    like = logical(1), checkSize = TRUE))
+
+    # if (length(cR) > 0) {
+    #     stop(cR)
+    # }
+    # Lmax cannot be more than graph size
+    # Validate the Lmax parameter
+    if (!missing(Lmax) &&
+        (Lmax > (numVertices <- igraph::vcount(EGG)))) {
```

Comment:
You are not following the algorithm I have outlined for you as pseudocode. I am stopping the review at this point. We need to talk.